pkg/ipamd/datastore/data_store.go

```diff
@@ -1197,21 +1117,6 @@ func (ds *DataStore) UnassignPodIPAddress(ipamKey IPAMKey) (e *ENI, ip string, d
 		ds.total, ds.assigned, ipamKey)
 
 	eni, availableCidr, addr := ds.eniPool.FindAddressForSandbox(ipamKey)
-	if addr == nil {
-		// This `if` block should be removed when the CRI
-		// migration code is finally removed.  Leaving a
-		// compile dependency here to make that obvious :P
-		var _ = checkpointMigrationPhase
-
-		// If the entry was discovered by querying CRI at
-		// restart rather than by observing an ADD operation
-		// directly, then we won't have captured the true
-		// networkname/ifname.
-		ds.log.Debugf("UnassignPodIPAddress: Failed to find IPAM entry under full key, trying CRI-migrated version")
-		ipamKey.NetworkName = backfillNetworkName
```

Comment:
this section cannot be removed. as customer upgrading from old CNI will still hit this branch. (since old pods is still have backfillNetworkName in their ipamd files)

Reply:
I see, so this comment means that once we guarantee existing pods were not created by VPC CNI version running checkpoint < 2, we can remove this `if` block?

Will that ever be true? We do only support customers upgrading one major version at a time, i.e. in 1.13, we will guarantee that we are upgrading from minimum 1.12, but pods could have been created before that upgrade.

Reply:
that will not be true. I don't know why the if statement were there(it's too optimistic :D) 

as long as we support the upgrade, there will always have old ipamd entries in the past with backfillNetworkName

Reply:
Got it, I'll just remove the comment then